cores/esp8266/Esp.cpp

```diff
@@ -109,6 +109,13 @@ extern "C" void esp_yield();
 
 void EspClass::deepSleep(uint64_t time_us, WakeMode mode)
 {
+    if (time_us > deepSleepMax()) // we need to prevent the esp8266 from not waking up from deepsleep
+    {
+       time_us = (deepSleepMax() - (round(deepSleepMax() * 5 / 100))); // 5% correction because of inaccurate timekeeping by the esp8266
```

Comment:
No need for round() here.  You're doing integer division already, so this takes the integer, promotes it to a double, then promotes deepSleepMax() to a double, subtracts them as doubles, then truncates it all back to an integer.  A shorter way which won't cause a program to include the whole floating point libs (if not being included already):
`time_us = (deepSleepMax() * 95L) / 100L;`
or if time_us can be > 2^24 (so the *95 but could overflow)
`time_us = deepSleepMax() - (deepSleepMax() / 20);`

I bring this up because there was much howling of pain when we included floating point support in printf with the latest release, as some folks binaries ended up being just a hair too big for OTA...